### PR TITLE
chore: 🎨  Add type hinting

### LINKED ISSUE #23: Use python built-in type hinting for functions.
Currently type hinting is not used in the codebase, leading to code like this (from `avmp/utils/ip_tools.py`)

```
def get_all_network_ips(network):
    """Given a network (0.0.0.0/24) return all possible ip addresses
    arg:
        network (str): IP network to extract IP's from
    return (list): IP's in given network
    """
    return [str(ip) for ip in IPNetwork(network)]
```

The alternative would be to do something like this
```
def get_all_network_ips(network: str) -> List[str]:
    """Given a network (0.0.0.0/24) return all possible ip addresses
    arg:
        network: IP network to extract IP's from
    return: IP's in given network
    """
    return [str(ip) for ip in IPNetwork(network)]
```

1. Code is more readable
2. Allows for integration of static typing tools like mypy to help with CI/CD
3. There is nothing enforcing or even suggesting the use of a string for the `network` argument (currently) unless a user looks up this function. 
4. Easy transition to using python's dataclasses library
5. Remove the visual clutter of typing from the docstring

### PR DESCRIPTION
**Pull request by Nate1729**
Resolves issue #23

### CODE REVIEW

File: avmp/utils/logging_utils.py [hunk 1]
```diff
@@ -5,11 +5,11 @@
 
 import os
 import sys
-import json
 import logging
+import json
```
Comment (RackReaver) [inline avmp/utils/logging_utils.py]: Why is this change needed?

**Review by RackReaver — CHANGES_REQUESTED** [review]

File: avmp/utils/logging_utils.py (hunk 1 above)
Comment (Nate1729) [reply avmp/utils/logging_utils.py]: It isn't I had to make another commit to redo the github workflow

**Comment by RackReaver** [comment]
Can you just run `isort` and `black` against these two files and then I will accept the pull.

**Comment by RackReaver** [comment]
I just added a GitHub workflow and changed the formatting of almost all the files, there is a conflict now that needs to be fixed. If you need assistance let me know.

**Review by RackReaver — APPROVED** [review]

**Comment by RackReaver** [comment]
@Nate1729 Please fix issue with `avmp/utils/logging_utils.py`
